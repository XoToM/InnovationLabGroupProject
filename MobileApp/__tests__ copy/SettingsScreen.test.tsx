import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import SettingsScreen from "../app/screens/settings"; // Adjust path as needed

describe("SettingsScreen", () => {
  it("renders the screen with header and title", () => {
    const { getByText } = render(<SettingsScreen />);
    expect(getByText("Settings")).toBeTruthy();
  });

  it("renders all accessibility switches", () => {
    const { getByText } = render(<SettingsScreen />);
    expect(getByText("Wheelchair user")).toBeTruthy();
    expect(getByText("Hearing loop")).toBeTruthy();
    expect(getByText("Guide dog owner")).toBeTruthy();
    expect(getByText("Social anxiety")).toBeTruthy();
  });

  it("renders all palette options", () => {
    const { getByText } = render(<SettingsScreen />);
    expect(getByText("Tol color palette")).toBeTruthy();
    expect(getByText("IBM color palette")).toBeTruthy();
  });

  it("toggles accessibility option on press", () => {
    const { getByText, getAllByA11yRole } = render(<SettingsScreen />);
    const wheelchairSwitch = getByText("Wheelchair user").parent?.parent?.find(
      (node) => node.props.accessibilityRole === "switch"
    );
    if (wheelchairSwitch) {
      fireEvent(wheelchairSwitch, "valueChange", true);
    }
    expect(getByText("Wheelchair user")).toBeTruthy(); // Still rendered
  });

  it("selects Tol palette", () => {
    const { getByText } = render(<SettingsScreen />);
    fireEvent.press(getByText("Tol color palette"));
    expect(getByText("Tol color palette")).toBeTruthy(); // Selected switch should reflect change
  });

  it("selects IBM palette", () => {
    const { getByText } = render(<SettingsScreen />);
    fireEvent.press(getByText("IBM color palette"));
    expect(getByText("IBM color palette")).toBeTruthy(); // Should remain true (default)
  });

  it("matches snapshot", () => {
    const tree = render(<SettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders icons for header and sections", () => {
    const { getAllByRole } = render(<SettingsScreen />);
    const icons = getAllByRole("button");
    expect(icons.length).toBeGreaterThanOrEqual(3); // home, close, and maybe others
  });

  it("toggles multiple accessibility options independently", () => {
    const { getByText } = render(<SettingsScreen />);
    fireEvent.press(getByText("Wheelchair user"));
    fireEvent.press(getByText("Hearing loop"));
    fireEvent.press(getByText("Social anxiety"));

    expect(getByText("Wheelchair user")).toBeTruthy();
    expect(getByText("Hearing loop")).toBeTruthy();
    expect(getByText("Social anxiety")).toBeTruthy();
  });

  it("does not crash when all switches are toggled", () => {
    const { getByText } = render(<SettingsScreen />);
    [
      "Wheelchair user",
      "Hearing loop",
      "Guide dog owner",
      "Social anxiety",
    ].forEach((label) => {
      fireEvent.press(getByText(label));
    });
    expect(getByText("Guide dog owner")).toBeTruthy();
  });
});
