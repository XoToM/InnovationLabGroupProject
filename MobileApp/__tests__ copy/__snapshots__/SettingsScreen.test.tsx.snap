// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen matches snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View>
     
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
       
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 10,
            "opacity": 1,
          }
        }
      >
         
      </View>
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Settings
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 10,
            "opacity": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "marginBottom": 5,
          "padding": 20,
          "width": "90%",
        }
      }
    >
       
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 10,
            "opacity": 1,
          }
        }
      >
         
      </View>
      <View
        style={
          {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 10,
            "padding": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
            }
          }
        >
           
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            Wheelchair user
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
            }
          }
        >
           
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            Hearing loop
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
            }
          }
        >
           
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            Guide dog owner
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
            }
          }
        >
           
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            Social anxiety
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "marginBottom": 5,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 10,
            "opacity": 1,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#f0f0f0",
            "borderRadius": 10,
            "padding": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
              "opacity": 1,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            Tol
             color palette
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 5,
              "opacity": 1,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={true}
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            IBM
             color palette
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
